src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -0,0 +1,832 @@
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(0));
+  message_to_send->set_json_message(std::string("message"));
+
+  EXPECT_CALL(mock_protocol_handler_, IsRPCServiceSecure(kConnectionKey))
+      .WillOnce(Return(false));
+  EXPECT_CALL(*mock_rpc_protection_manager_, IsInEncryptionNeededCache(_, _))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_protocol_handler_, SendMessageToMobileApp(_, _, _)).Times(0);
+  message_handler->Handle(
+      am::rpc_service::impl::MessageToMobile(message_to_send, false));
+}
+
+TEST_F(RPCServiceImplTest, Handle_HappyPathWithoutClosingSession) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  auto message_to_send = std::make_shared<am::Message>(
+      protocol_handler::MessagePriority::kDefault);
+  message_to_send->set_protocol_version(
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_1);
+  message_to_send->set_connection_key(kConnectionKey);
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(2));
+  message_to_send->set_json_message(std::string("message"));
+
+  EXPECT_CALL(mock_protocol_handler_, IsRPCServiceSecure(kConnectionKey))
+      .WillOnce(Return(true));
+  ON_CALL(mock_app_mngr_, application(_)).WillByDefault(Return(nullptr));
+  EXPECT_CALL(*mock_rpc_protection_manager_, CheckPolicyEncryptionFlag(_, _, _))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_protocol_handler_, SendMessageToMobileApp(_, _, _))
+      .WillOnce(Return());
+  EXPECT_CALL(*mock_rpc_protection_manager_,
+              RemoveFromEncryptionNeededCache(_, _))
+      .WillOnce(Return());
+  message_handler->Handle(
+      am::rpc_service::impl::MessageToMobile(message_to_send, false));
+}
+
+TEST_F(RPCServiceImplTest, Handle_HappyPathWithClosedSession) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  auto message_to_send = std::make_shared<am::Message>(
+      protocol_handler::MessagePriority::kDefault);
+  message_to_send->set_protocol_version(
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_1);
+  message_to_send->set_connection_key(kConnectionKey);
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(2));
```

Comment:
Magic number. Please provide concrete enum value or make an appropriate constant.

Reply:
Done in commit d5ec8240cfac58e5ea9f7f9ed13f8127a02f94b9